### PR TITLE
Added commands for Microsoft.DevTestLab provider.

### PR DESCRIPTION
**Pull request by shichaoh**
- Added commands for Microsoft.DevTestLab provider.
- Also added some tests and recordings. Will add more tests in future commits.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shichaoh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shichaoh). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/devtestlabs/devTestLabs._js [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+  //   Auto-shutdown policy 
+  var autoShutdownPolicyCommands = labCommands.category('auto-shutdown-policy');
+  
+  autoShutdownPolicyCommands.command('set')
+    .description($('Set virtual machine auto-shutdown policy.'))
+    .option('-g, --resource-group <resource-group>', $('the resource group name of the lab'))
+    .option('-l, --lab-name <lab-name>', $('the lab name'))
+    .option('-w, --weekly [weekly]', $('weekdays in a weekly schedule. example value: Monday,Friday'))
+    .option('-d, --daily', $('set as a daily schedule'))
+    .option('-t, --time <time>', $('the time to shutdown virtual machines, example format: 1905 (it means 7:05pm)'))
```
Comment (amarzavery) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: So if some one says "-w 'Monday, Friday' and -t 2000 -z PST" - Does this mean that the vm will be shut down on all week days at 8.00 pm PST.

What will happen when the user just provides "-t 2000 -z PST"? Will it just shut down once on the following day or every day?

Please update the help to clarify such scenarios.

File: lib/commands/arm/devtestlabs/devTestLabs._js [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+  
+  autoStartPolicyCommands.command('set')
+    .description($('Set virtual machine auto-start policy.'))
+    .option('-g, --resource-group <resource-group>', $('the resource group name of the lab'))
+    .option('-l, --lab-name <lab-name>', $('the lab name'))
+    .option('-w, --weekly [weekly]', $('weekdays in a weekly schedule. example value: Monday,Friday'))
+    .option('-d, --daily', $('set as a daily schedule'))
+    .option('-t, --time <time>', $('the time to start virtual machines, example value: 1905 (it means 7:05pm '))
+    .option('-z, --time-zone-id [time-zone-id]', $('time zone id, default to UTC'))
+    .option('-s, --subscription <id>', $('the subscription identifier'))
```
Comment (amarzavery) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: same as above for shutdown

File: test/commands/arm/devtestlabs/arm.devtestlabs-tests.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  describe('set vms per lab policy', function () {
+    it('Set vms per lab policy command should work', function (done) {
+      var vmCount = 10000;
+      suite.execute('lab vms-per-lab-policy set --resource-group %s --lab-name %s --vm-count %s --subscription %s', testResourceGroup, testLabName, vmCount, testSubscription, function (result) {
+        result.exitStatus.should.be.equal(0);
+        done();
+      });
+    });
+
+    it('get vms per lab policy command should work', function (done) {
```
Comment (amarzavery) [inline test/commands/arm/devtestlabs/arm.devtestlabs-tests.js]: What about other commands? We need atleast one test for every command.

**Comment by amarzavery** [comment]
@shichaoh Overall the PR looks good. Please address the review feedback.

File: lib/commands/arm/devtestlabs/devTestLabs._js (hunk 1 above)
Comment (shichaoh) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: Thanks! Updated the command to make it simper to match with what our Ibiza UX provides. Also added more description to clarify. Removed the time-zone-id option as there is no standard to get time zone id, it is always defaults to UTC in this very first version.

File: lib/commands/arm/devtestlabs/devTestLabs._js (hunk 2 above)
Comment (shichaoh) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: Updated. now the syntax are simpler.

File: test/commands/arm/devtestlabs/arm.devtestlabs-tests.js (hunk 3 above)
Comment (shichaoh) [inline test/commands/arm/devtestlabs/arm.devtestlabs-tests.js]: Thanks! added two tests for every command.

File: lib/commands/arm/devtestlabs/devTestLabs._js [hunk 4]
```diff
@@ -349,16 +338,15 @@ exports.init = function (cli) {
...
   
   autoStartPolicyCommands.command('set')
     .description($('Set virtual machine auto-start policy.'))
     .option('-g, --resource-group <resource-group>', $('the resource group name of the lab'))
     .option('-l, --lab-name <lab-name>', $('the lab name'))
-    .option('-w, --weekly [weekly]', $('weekdays in a weekly schedule. example value: Monday,Friday'))
-    .option('-d, --daily', $('set as a daily schedule'))
-    .option('-t, --time <time>', $('the time to start virtual machines, example value: 1905 (it means 7:05pm '))
-    .option('-z, --time-zone-id [time-zone-id]', $('time zone id, default to UTC'))
+    .option('-w, --weekly <weekly>.', $('selected weekdays to start machines. example: Monday,Friday means start machines on Monday and Friday.'))
```
Comment (amarzavery) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: @shichaoh - you removed the weekly option from shutdown but kept it in autostart. Was this intentional? Just verifying.

Comment (shichaoh) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: Yes, that is intentionally. For the main scenario, shutdown will always be daily. For autostart, user can choose specific days. Thanks for checking!

Comment (amarzavery) [inline lib/commands/arm/devtestlabs/devTestLabs._js]: sounds good!

**Pull request merged**